backend/app/auth_handlers.go

```
@@ -0,0 +1,246 @@
+package app
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/codescalersinternships/Flyspray/internal"
+	"github.com/codescalersinternships/Flyspray/models"
+	"gorm.io/gorm"
+
+	"github.com/gin-gonic/gin"
+)
+
+func (a *App) Signup(ctx *gin.Context) (interface{}, Response) {
+	var user models.User
+
+	err := json.NewDecoder(ctx.Request.Body).Decode(&user)
+
+	if err != nil {
+		return nil, BadRequest(err)
```

you should return an error that the user will understand. `input data is invalid` for example

apply for all